ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx

```diff
@@ -170,6 +257,38 @@ function renderFilteredNode(node: {count: number} & dagre.Node, onClearFilter: (
     );
 }
 
+function renderGroupedNodes(props: ApplicationResourceTreeProps, node: {count: number} & dagre.Node & ResourceTreeNode) {
+    const indicators = new Array<number>();
+    const groupdedNodeIds = node.groupdedNodeIds;
+    let count = Math.min(node.count - 1, 3);
+    while (count > 0) {
+        indicators.push(count--);
+    }
+    return (
+        <React.Fragment>
+            <div className='application-resource-tree__node' style={{left: node.x, top: node.y, width: node.width, height: node.height}}>
+                <div className='application-resource-tree__node-kind-icon'>
+                    <ResourceIcon kind={node.kind} />
+                    <br />
+                    <div className='application-resource-tree__node-kind'>{ResourceLabel({kind: node.kind})}</div>
+                </div>
+                <div className='application-resource-tree__node-content-wrap-overflow'>
+                    <a className='application-resource-tree__node-title' onClick={() => props.onGroupdNodeClick && props.onGroupdNodeClick(groupdedNodeIds)}>
+                        click to show details of {node.count} collapsed {node.kind}
+                    </a>
+                </div>
+            </div>
+            {indicators.map(i => (
```

Comment:
Hi Yi, help me to understand what this extra div is for. Do we need it?

Reply:
If you mean the block starts with `indicators.map`, it is to demonstrate multiple nodes are collapsed as one group. Please refer the below screenshot.

Without the div
![Screen Shot 2021-11-24 at 2 29 22 PM](https://user-images.githubusercontent.com/26255262/143302113-379d047f-85e7-4bde-867e-6c421333bec6.png)
With the div
![Screen Shot 2021-11-24 at 2 30 43 PM](https://user-images.githubusercontent.com/26255262/143302117-30e75b18-42ef-44e8-9514-68377a4d2573.png)